src/listmode.cpp

```diff
@@ -140,8 +140,9 @@ unsigned int ListModeBase::GetLowerLimit()
 	unsigned int limit = UINT_MAX;
 	for (const auto& chanlimit : chanlimits)
 	{
+		// This cast is safe as we start at UINT_MAX and reduce from there.
 		if (chanlimit.limit < limit)
-			limit = chanlimit.limit;
+			limit = static_cast<unsigned int>(chanlimit.limit);
```

Comment:
The method returns unsigned long but you cast down to unsigned int here. Is there a reasonf ro that?

Reply:
`GetLowerLimit` (this method) returns `unsigned int` still, which is safe per the above comment. `GetLimit` returns `unsigned long` with my changes in this PR as there was no safe way to keep it as `int`. This is inconsistent though so if you think this (`GetLowerLimit`) should return `unsigned long` too, i can make that change.

Reply:
Ah yeah, you can alter this method too.